primedev/engine/gl_matsysiface.cpp

```diff
@@ -0,0 +1,50 @@
+#include "materialsystem/cmaterialglue.h"
+
+CMaterialGlue* (*GetMaterialAtCrossHair)();
+
+AUTOHOOK_INIT()
+
+AUTOHOOK(CC_mat_crosshair_printmaterial_f, engine.dll + 0xB3C40, void, __fastcall, (const CCommand& args))
+{
+	CMaterialGlue* pMat = GetMaterialAtCrossHair();
+
+	if (!pMat)
+	{
+		spdlog::error("Not looking at a material!");
+		return;
+	}
+
+	std::function<void(CMaterialGlue * pGlue, const char* szName)> fnPrintGlue = [](CMaterialGlue* pGlue, const char* szName)
```

Comment:
There isn't really a need for std::function here given the slight extra overhead of doing so. Just storing the lambda using `auto` is mildly faster.

Absolutely a nitpick though, I doubt that this will actually matter enough for us to care.
```suggestion
	auto fnPrintGlue = [](CMaterialGlue* pGlue, const char* szName) -> void
```

Reply:
Took this from r5sdk, this isn't performance critical code so ill just leave it as is. When i revisit the other `mat_crosshair` cvars ill look into making this nicer <3